### PR TITLE
Migrate to new Azure SDK packages

### PR DESCRIPTION
**Pull request by RMacfarlane**
https://github.com/microsoft/vscode-azure-account/issues/140

Migrate to using the new Azure SDK packages internally, but continue to expose the old credentials object in the API for backcompatability. Introduces a new "credentialsV2" property corresponding to the new credentials object

### CODE REVIEW

**Comment by chrmarti** [comment]
@RMacfarlane Actually everything works when running out-of-source, only when installing the packaged VSIX it somehow didn't. Can you give that a try?

File: src/cloudConsole.ts [hunk 1]
```diff
@@ -380,7 +380,7 @@ export function createCloudConsole(api: AzureAccount, reporter: TelemetryReporte
 		// Additional tokens
 		const [graphToken, keyVaultToken] = await Promise.all([
 			tokenFromRefreshToken(session.environment, result.token.refreshToken, session.tenantId, session.environment.activeDirectoryGraphResourceId),
-			tokenFromRefreshToken(session.environment, result.token.refreshToken, session.tenantId, `https://${session.environment.keyVaultDnsSuffix.substr(1)}`)
+			tokenFromRefreshToken(session.environment, result.token.refreshToken, session.tenantId, `https://${session.environment.keyVaultDnsSuffix!.substr(1)}`)
```
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/cloudConsole.ts]: Yeah, I just tried without a KeyVault token and it seemed to work fine without it. I'll change this to make the keyVault token optional

File: src/azure-account.api.d.ts [hunk 2]
```diff
@@ -27,10 +28,11 @@ export interface AzureAccount {
 }
 
 export interface AzureSession {
-	readonly environment: AzureEnvironment;
+	readonly environment: Environment;
 	readonly userId: string;
 	readonly tenantId: string;
 	readonly credentials: ServiceClientCredentials;
+	readonly credentialsV2: DeviceTokenCredentials;
```
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/azure-account.api.d.ts]: Changed to `TokenCredentialsBase`, which seems to be the most general interface for the new SDK

**Comment by RMacfarlane** [comment]
Was it sign in that didn't work? I've just tried packaging and running it as a VSIX and have been able to sign in and open a cloud console.

**Comment by chrmarti** [comment]
I see the following in the console when running with the VSIX, but not when running with `yarn watch`. Do you see that? Maybe the minifier somehow corrupts things.
```
[Extension Host] TypeError: Expected signal to be an instanceof AbortSignal
	at new B (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:12277)
	at /Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:14625
	at new Promise (<anonymous>)
	at G (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:14589)
	at t.<anonymous> (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:76748)
	at /Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:1768
	at Object.next (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:1873)
	at /Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:811
	at new Promise (<anonymous>)
	at a (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:548)
	at t.fetch (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:76672)
	at t.<anonymous> (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:364:74931)
	at /Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:1768
	at Object.next (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:1873)
	at a (/Users/chrmarti/Development/repos/vscode-azure-account/dist/extension.js:16:604)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by chrmarti** [comment]
This appears to be https://github.com/node-fetch/node-fetch/issues/784. Just not throwing the error by manually removing it makes it work. The solution on that issue was to not minify class names. How did you build the VSIX avoiding this issue?

**Comment by RMacfarlane** [comment]
Hmm, I'm just building with `vsce package`, but I still don't see that in the console

**Comment by chrmarti** [comment]
What do you get for regex searching for `class.*EventTarget` in `dist/extension.js` after compiling with `npm run vscode:prepublish`? My first match is the minified AbortSignal class:
```js
class i extends r.EventTarget{constructor(){throw super(),new TypeError("AbortSignal cannot be constructed directly")} ...
```

Its name has been minified to `i`. 

Searching for `"AbortSignal"` finds me the check that no longer works:
```js
"AbortSignal"!==t.constructor.name)}(a))throw new TypeError("Expected signal to be an instanceof AbortSignal");
```

Do you have these too? I'd expect so. Maybe you just don't run through this code for some reason?

**Comment by RMacfarlane** [comment]
Ah, thanks, I do see the same minification in the output, so I must just not be hitting that code path for some reason.

I pushed a change to turn off minification, which unfortunately doubles the size of the `.js` file to 4MB, but retains the class name.

**Pull request merged**